src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
```

Comment:
Yes, I guess it's ok, but we should probably change the message to simply...

    throw new IllegalStateException("The 'IntegrationFlow' 

Also `s/dependant/dependent/`.